### PR TITLE
search for micromamba in more locations

### PR DESCRIPTION
**Pull request by savingoyal**

### CODE REVIEW

File: metaflow/plugins/pypi/bootstrap.py [hunk 1]
```diff
@@ -116,7 +121,9 @@
         # Install PyPI packages.
         cmds.extend(
             [
-                f"""./micromamba run --prefix {prefix} pip --disable-pip-version-check install --root-user-action=ignore --no-compile {pypi_pkgs_dir}/*.whl"""
+                f"""set -e;
+                export PATH=$PATH:$(pwd)/micromamba;
```
Comment (romain-intel) [inline metaflow/plugins/pypi/bootstrap.py:125]: ditto here.

File: metaflow/plugins/pypi/conda_environment.py [hunk 2]
```diff
@@ -314,6 +314,7 @@ def bootstrap_commands(self, step_name, datastore_type):
                 'python -m metaflow.plugins.pypi.bootstrap "%s" %s "%s" linux-64'
                 % (self.flow.name, id_, self.datastore_type),
                 "echo 'Environment bootstrapped.'",
+                "export PATH=$PATH:$(pwd)/micromamba",
```
Comment (romain-intel) [inline metaflow/plugins/pypi/conda_environment.py:317]: Is it needed later?

File: metaflow/plugins/pypi/micromamba.py [hunk 3]
```diff
@@ -277,10 +277,12 @@ def _install_micromamba(installation_location):
     try:
         subprocess.Popen(f"mkdir -p {installation_location}", shell=True).wait()
         # https://mamba.readthedocs.io/en/latest/micromamba-installation.html#manual-installation
+        # requires bzip2
         result = subprocess.Popen(
-            f"curl -Ls https://micro.mamba.pm/api/micromamba/{platform}/latest | tar -xvj -C {installation_location} bin/micromamba",
+            f"curl -Ls https://micro.mamba.pm/api/micromamba/{platform}/latest | tar -xvj -C {installation_location} bin/micromamba >",
```
Comment (romain-intel) [inline metaflow/plugins/pypi/micromamba.py:282]: what's with the redirect?

Comment (savingoyal) [reply metaflow/plugins/pypi/micromamba.py:282]: good catch - it's spurious. i will remove

File: metaflow/plugins/pypi/conda_environment.py (hunk 2 above)
Comment (savingoyal) [reply metaflow/plugins/pypi/conda_environment.py:317]: yep - to placate `micromamba.py`

File: metaflow/plugins/pypi/bootstrap.py (hunk 1 above)
Comment (savingoyal) [reply metaflow/plugins/pypi/bootstrap.py:125]: yes - these are executed within different shells right now - they can be moved into a single shell command as well as a quick follow up.

File: metaflow/plugins/pypi/bootstrap.py [hunk 4]
```diff
@@ -87,18 +87,23 @@
...
             fi;
         fi""",
         # Create a conda environment through Micromamba.
-        f'''tmpfile=$(mktemp);
+        f'''set -e;
+        tmpfile=$(mktemp);
         echo "@EXPLICIT" > "$tmpfile";
         ls -d {conda_pkgs_dir}/*/* >> "$tmpfile";
-        ./micromamba create --yes --offline --no-deps --safety-checks=disabled --no-extra-safety-checks --prefix {prefix} --file "$tmpfile";
+        export PATH=$PATH:$(pwd)/micromamba;
```
[the remark this answers is not in the document]
Comment (savingoyal) [reply metaflow/plugins/pypi/bootstrap.py:105]: https://github.com/Netflix/metaflow/pull/1581#discussion_r1353059942

**Pull request merged**